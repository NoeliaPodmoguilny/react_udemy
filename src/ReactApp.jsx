// props ==> properties son las propiedades que le pasamos al component (Fn), esto es un objeto. para evitar tener que escribir props.loquesea... tenemos que desestructurar la props {loquesea} 
//las props vienen del component padre (main.jsx) al componente hijo (reactapp.jsx)
import PropTypes from 'prop-types'

export const ReactApp = ({title, subtitle, name}) => {

    // console.log(props);
    
    return (
        <>
        <h1>{ title }</h1>
        <h3>{ subtitle }</h3>
        <p>{ name }</p>
        </>
    )
}
//definir propiedades por defecto:
ReactApp.defaultProps = {
    title: 'no hay titulo',
    subtitle: 'no hay subtitulo',
    name: 'Noelia'
}
//entran antes que los propsTypes
//manejar los tipos y otras condiciones (el prop-types se debe instalar!!) Al usar typescript no es necesario
ReactApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string
}
