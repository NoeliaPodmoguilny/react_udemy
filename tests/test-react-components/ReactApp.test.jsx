import { render } from "@testing-library/react"
import { ReactApp } from "../../src/ReactApp.jsx"

describe('Prueba de <ReactApp />', () => {

    // test('Debe de hacer match con el snapshot', () => {

    //     const title = 'Hola, soy Noelia'
    //     const { container } = render( <ReactApp title={title}/>)
    //     // console.log(container);
    //     expect(container).toMatchSnapshot()
    // })

    test('debe de mostrar el titulo en un h1', () => {

        const title = 'Hola, soy Noelia'
        const { container, getByText, getByTestId } = render( <ReactApp title={title}/>)
        
        expect(getByText(title)).toBeTruthy()

        // const h1 = container.querySelector('h1')
        // expect(h1.innerHTML).toContain(title)

        expect(getByTestId('test-title').innerHTML).toContain(title)
    
    })
})