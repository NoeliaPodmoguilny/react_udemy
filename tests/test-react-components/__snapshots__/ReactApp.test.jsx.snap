// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de <ReactApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, soy Noelia
  </h1>
  <h3>
    no hay subtitulo
  </h3>
  <p>
    Noelia
  </p>
</div>
`;
